File: go/spongify_test.go
package main

import (
	"testing"

	"github.com/bxcodec/faker/v3"
)

func getSpongetext() (input string, output string) {
	input = faker.Sentence()
	output = spongify(input)
	return input, output
}

func TestNotUndefined(t *testing.T) {
	_, output := getSpongetext()
	isNotBlank := output != ""
	if isNotBlank == false {
		t.Error("spongetext() got blank, expected string")
	}
}

func TestShouldHaveSameLength(t *testing.T) {
	input, output := getSpongetext()
	isSameLength := len(output) == len(input)
	if isSameLength == false {
		t.Error("result does not have the same length as input")
	}
}

File: ts/src/spongify.test.ts
import { faker } from '@faker-js/faker';
import { describe, expect, it } from 'vitest';
import spongify from './spongify';

describe('Spongify test', () => {
  const input = faker.lorem.sentence(15);
  const output = spongify(input);

  it('should spongify text properly', () => {
    expect(output).toBeDefined();
  });

  it('should generate a string output', () => {
    expect(typeof output).toBe('string');
  });

  it('should have the same length as the input', () => {
    expect(output.length).toEqual(input.length);
  });
});
